internal/contact/email.go

```diff
@@ -0,0 +1,43 @@
+package contact
+
+import (
+	"context"
+	"fmt"
+	"github.com/CollActionteam/collaction_backend/internal/constants"
+	"github.com/CollActionteam/collaction_backend/internal/models"
+)
+
+const separator = "### app version:"
+
+type EmailRepository interface {
+	Send(ctx context.Context, data models.EmailData) error
+}
+
+type ConfigManager interface {
+	GetParameter(name string) (string, error)
+}
+
+type Contact struct {
```

Comment:
Does Contact is a domain object? Why does it receive an EmailRepository?

Reply:
yes, choosing the email provider is not the Service concern, it dictates that to send an email it needs a method like the defined one, and the user of the service (in this case sam controller or HTTP controller) needs to provide it.
by doing that we separate our concerns.